lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -191,7 +191,7 @@ class _MetricsEndpointDict(TypedDict):
 LIBAPI = 0
 LIBPATCH = 3
 
-PYDEPS = ["cosl", "pydantic"]
+PYDEPS = ["cosl", "pydantic<2"]
```

Comment:
Is there any limitation why we need pydantic<2? I would only do this if there's no other option available. Can't we adapt the charm code?

Reply:
The build fails, it's a pb in charmcraft, no in the lib itself

Reply:
BTW the PR has landed on cos_agent lib, so we can fetch the latest version of the lib instead of changing it ourselves